HARJA-1337 YHA.xsd skeeman päivitys

jarnovayrynen opened the pull request.

jarnovayrynen committed:
Päivitä YHA-skeemaa

YHA:n kuvaus muutoksista:
"Ei mitään isoja rakenteellisia muutoksia. Lähinnä 99 - ei tietoa poistoja, yksi typon korjaus ja muutamia lisäyksiä. "paallyste" eli teidän hakuun tulevaan vastaukseen liittyvä pinnan nykyinen pinnoitus ja suunnitellun päällystetyön pinta. Puu ja teräs liittyvät nykypintaan (meille tulee lähtötiedoissa näitä arvoja), muut lisäykset tähän vaan sisältää jo teiltä meillepäin tulevia koodeja (yhdenmukaistetaan ne tähänkin). "alustamenetelma" sai koodeilla 5x olevat lisäykset asiakkaan pyynnöstä. "sideainetyyppi" korjattu Katrin pyynnöstä kalkkifilleri joka on väärässä paikassa ja lisätty 27 - uusiomateriaali."

jarnovayrynen commented on resources/xsd/yha/yha.xsd:
```diff
@@ -552,7 +552,7 @@
                             <xs:annotation>
                                 <xs:documentation>
-                                    Masuunikuomajauhe
+                                    Masuunikuonajauhe
```
Harjan päässä OK. Jatkossa merkkaan jokaisen muutoksen OK sitten kun on joko tarkistettu tai korjattu OK:ksi.

jarnovayrynen commented on resources/xsd/yha/yha.xsd:
```diff
@@ -735,6 +735,20 @@
                     </xs:documentation>
                 </xs:annotation>
             </xs:enumeration>
+            <xs:enumeration value="3">
```
nämä OK, lisätty tauluun pot2_mk_massatyyppi  ja analytiikkasanomaan

jarnovayrynen commented on resources/xsd/yha/yha.xsd:
```diff
@@ -1083,9 +1127,29 @@
                 </xs:annotation>
             </xs:enumeration>
-            <xs:enumeration value="99">
+            <xs:enumeration value="50">
```
Nämä ok, laitettu pot2_mk_alusta_toimenpide tauluun ja analytiikkasanomaan

jarnovayrynen committed:
HARJA-1337 YHA.xsd skeeman päivitys

Päivitä tietokantaan skeemoja vastaavat koodistot. Kaikkea POT1-lomakkeeseen liittyvää ei tässä yritetäkään päivittää, koska sitä ei enää moneen vuoteen ole aktiivisesti käytetty eikä sinne kirjata uusia arvoja.

Myös käytöstä poistuneen vanhan POT1 rajapinnan json skeema opoistetaan tässä turhana, kun oli jäänyt roikkumaan.

solita-staskila commented on tietokanta/src/main/resources/db/migration/V1_1164__.sql:
```diff
@@ -0,0 +1,36 @@
+
+ALTER TABLE pot2_mk_massatyyppi ADD COLUMN poistettu BOOLEAN DEFAULT FALSE NOT NULL;
+ALTER TABLE pot2_mk_massatyyppi ADD COLUMN muokkaaja INTEGER REFERENCES kayttaja (id) DEFAULT NULL;
+ALTER TABLE pot2_mk_massatyyppi ADD COLUMN muokattu TIMESTAMP DEFAULT NULL;
```
Onko taulussa olemassa jo luotu, luoja?

jarnovayrynen replied:
Ei ole. Nuo on olleet vain meidän migraatiossa luomia, eikä näille ole esim kälissä mitään muokkausta. Tämä olisi ensimmäinen "poisto" niin siinä mielessä ajattelin että tekisimme tarpeen mukaan näitä metatietoja jos koodistot muuttuu. se "luonti" on tavallaan tallessa siellä migraatiotiedoston päivämäärässä gitissä.

solita-staskila reviewed (approved):
1337

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HARJA-1337

jarnovayrynen committed:
Päivitä migraatiota, älä luokaan massatyyppejä

Yha-skeemaan lisättyjen päällysteiden on tarkoitus olla vain hakurajapinnassa, ei enää Harjasta YHA:an päin, joten ei lisätä niitä massatyyppi-tauluun.

jarnovayrynen committed:
Päivitä json skeemat
